ango_user/app/routers/user.py

```diff
@@ -32,16 +25,13 @@ def register_user(create_request: UserCreateRequest) -> UserDb:
     return result
 
 
-@user_router.delete("/")
-def delete_user(delete_request: UserDeleteRequest, access_token: str = Cookie(...)):
-    try:
-        payload = jwt.decode(access_token, settings.SECRET_KEY, algorithms=[settings.ALGORITHM])
-        token_data = TokenData.parse_obj(payload)
-        if token_data.email != delete_request.email:
-            raise AuthException(message="Invalid email detected")
+@user_router.get("/{user_email}", dependencies=[Depends(authorize_service)])
+def get_user_by_email(user_email: str) -> User:
+    user = user_service.get_user_by_email(user_email)
+    return user
 
-    except (JWTError, ValidationError):
-        raise AuthException(message="Invalid token detected")
 
-    user_service.delete(delete_request)
+@user_router.delete("/", dependencies=[Depends(authorize_user)])
+def delete_user(request: UserDeleteRequest):
+    user_service.delete(request)
```

Comment:
No checking for a authenticated UserA trying to send details of UserB in the request and causing a subsequent delete?

Reply:
Check is there, the dependency authorize_user will do that

Reply:
Cool, my bad.